mergify_engine/github_types.py

```diff
@@ -137,8 +137,8 @@ class GitHubIssue(GitHubIssueOrPullRequest):
     "has_hooks",
 ]
 
-GitHubPullRequestId = typing.NewType("GitHubPullRequestId", GitHubIssueId)
-GitHubPullRequestNumber = typing.NewType("GitHubPullRequestNumber", GitHubIssueNumber)
+GitHubPullRequestId = typing.NewType("GitHubPullRequestId", int)
+GitHubPullRequestNumber = typing.NewType("GitHubPullRequestNumber", int)
```

Comment:
The problem I think is that if you have a method like
```python
def retrieve_comments(issue_id: GitHubIssueId):
```
and that works on both issues and PRs, then you can't pass a `GitHubPullRequestId` to it because they are different.

Reply:
This method should take `typing.Union[GitHubIssueId, GitHubPullRequestId]` then.